backpack/extensions/saved_quantities.py

```diff
@@ -0,0 +1,39 @@
+"""Class for saving backpropagation quantities."""
+from typing import Dict, Union
+
+from torch import Tensor
+
+
+class SavedQuantities:
+    """Implements interface to save backpropagation quantities."""
+
+    def __init__(self):
+        """Initialization."""
+        self._saved_quantities: Dict[int, Tensor] = {}
+
+    def save_quantity(self, key: int, quantity: Tensor) -> bool:
+        """Saves the quantity under the specified key.
+
+        Args:
+            key: id of grad_input[0]
+            quantity: tensor to save
+
+        Returns:
+            whether save was successful
+        """
+        if key in self._saved_quantities:
+            return False
```

Comment:
Raise a NotImplementedError instead, because this is not covered in tests.
Expected bahaviour is to sum the quantities
Needed for two cases:
- `retain_graph=True`
- ResNets' branching points

@f-dangel Do we need tests with `retain_graph=True`?
I suggest to leave it as a NotImplementedError and that I create an issue.
```suggestion
        if key in self._saved_quantities:
            raise NotImplementedError
```

Reply:
The issue is not precise any more. It is more about the clean-up after a backward hook. A wrote a corresponding TODO in the code.